lib/sanford/template_engine.rb

```diff
@@ -30,12 +30,16 @@ def ==(other_engine)
 
   class NullTemplateEngine < TemplateEngine
 
-    def render(name, service_handler, locals)
-      template_file = self.source_path.join(name).to_s
-      unless File.exists?(template_file)
-        raise ArgumentError, "template file `#{template_file}` does not exist"
+    def render(template_name, service_handler, locals)
+      paths = Dir.glob(self.source_path.join("#{template_name}*"))
+      if paths.size > 1
+        raise ArgumentError, "#{template_name.inspect} matches more than one " \
+                             "file, consider using a more specific template name"
+      end
+      if paths.size < 1
+        raise ArgumentError, "a template named #{template_name.inspect} does not exist"
```

Comment:
@jcredding I'm using this error message in Deas, FYI: "a template file named with ...".  Just trying to be specific about how the template lookups work.

See also:
* https://github.com/redding/deas-erubis/pull/27/files#diff-76e3cfcec93d9c7ca3a112a5bf6c9a45R51
* https://github.com/redding/nm/pull/17/files#diff-461c530e59120e9d700d5735752898e2R33